samples/azure-quantum/variational-quantum-eigensolver/VQE-qiskit-hydrogen-ionq-sim.ipynb

```diff
@@ -0,0 +1,520 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "9b4ce39a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "driver = PySCFDriver(atom='H .0 .0 -0.3625; H .0 .0 0.3625', unit=UnitsType.ANGSTROM, charge=0, spin=0, basis='sto3g')\n",
+    "molecule = driver.run()\n",
+    "num_particles = molecule.num_alpha + molecule.num_beta\n",
+    "qubitOp = FermionicOperator(h1=molecule.one_body_integrals, h2=molecule.two_body_integrals).mapping(map_type='parity')\n",
+    "qubitOp = Z2Symmetries.two_qubit_reduction(qubitOp, num_particles)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "50174a98",
+   "metadata": {},
+   "source": [
+    "## 1. Simulate locally\n",
+    "\n",
+    "We will simulate the program locally using the Aer simulator. We create a QuantumInstance with a noise model, generated using a mock device \"FakeVigo\" with noise characteristics."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "737c1000",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from qiskit.test.mock import FakeVigo\n",
+    "from qiskit.providers.aer import AerSimulator\n",
+    "from qiskit.providers.aer import QasmSimulator\n",
+    "from qiskit.providers.aer.noise import NoiseModel\n",
+    "\n",
+    "backend = AerSimulator()\n",
+    "device_backend = FakeVigo()\n",
+    "device = QasmSimulator.from_backend(device_backend)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f9939a47",
+   "metadata": {},
+   "source": [
+    "Then, run the simulation using the VQE class."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d3ceaa01",
+   "metadata": {},
+   "source": [
+    "### Simulate locally with noise and error mitigation"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "2f799a78",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Exact Result: [-1.13722138]\n",
+      "VQE Result on noisy simulator: -1.123499771168513\n"
+     ]
+    }
+   ],
+   "source": [
+    "coupling_map = device.configuration().coupling_map\n",
+    "noise_model = NoiseModel.from_backend(device)\n",
+    "basis_gates = noise_model.basis_gates\n",
+    "\n",
+    "quantum_instance = QuantumInstance(backend=backend, \n",
+    "                                   shots=8192, \n",
+    "                                   noise_model=noise_model, \n",
+    "                                   coupling_map=coupling_map,\n",
+    "                                   measurement_error_mitigation_cls=CompleteMeasFitter,\n",
+    "                                   cals_matrix_refresh_period=30)\n",
+    "\n",
+    "exact_solution = NumPyEigensolver(qubitOp).run()\n",
+    "print(\"Exact Result:\", np.real(exact_solution.eigenvalues) + molecule.nuclear_repulsion_energy)\n",
+    "optimizer = SPSA(maxiter=100)\n",
+    "var_form = EfficientSU2(qubitOp.num_qubits, entanglement=\"linear\")\n",
```

Comment:
I'll add in a reference to the RyRz ansatz. As per the Qiskit documentation:
> As the effects of noise increase as the number of two qubit gates circuit depth increase, we use a heuristic variational form (RYRZ) rather than UCCSD as RYRZ has a much shallower circuit than UCCSD and uses substantially fewer two qubit gates.